backend/src/main/java/com/github/chuettenrauch/mixifyapi/security/oauth2/AuthorizationRequestRepositoryDecorator.java

```diff
@@ -0,0 +1,52 @@
+package com.github.chuettenrauch.mixifyapi.security.oauth2;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.oauth2.client.web.AuthorizationRequestRepository;
+import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationRequest;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+@RequiredArgsConstructor
+public class AuthorizationRequestRepositoryDecorator implements AuthorizationRequestRepository<OAuth2AuthorizationRequest> {
+
+    private static final String REDIRECT_URL_PARAM = "redirect_url";
+    private final AuthorizationRequestRepository<OAuth2AuthorizationRequest> authorizationRequestRepository;
```

Comment:
Ist das hier ein MongoRepository? Wenn ja, muss hier nicht noch ein String in die eckigen Klammern?

Reply:
Nee, das is son eigending von Spring Security OAuth2. 
Da wird der ursprüngliche Login Request zwischengespeichert.